controllers/ready_controller.go

```diff
@@ -322,6 +323,25 @@ func (rc *readinessChecker) check(ctx context.Context, c *redskyv1beta1.Readines
 		}
 	}
 
+	// If there are no items to check, try to provide a useful message
+	if len(ul.Items) == 0 {
+		var missingTargetMsg strings.Builder
+		missingTargetMsg.WriteString("No matching resources found")
+		missingTargetMsg.WriteString("; apiVersion=")
+		missingTargetMsg.WriteString(c.TargetRef.APIVersion)
+		missingTargetMsg.WriteString("; kind=")
+		missingTargetMsg.WriteString(c.TargetRef.Kind)
+		if c.TargetRef.Name != "" {
+			missingTargetMsg.WriteString("; name=")
+			missingTargetMsg.WriteString(c.TargetRef.Name)
+		}
+		if c.Selector != nil {
+			missingTargetMsg.WriteString("; labels=")
+			missingTargetMsg.WriteString(metav1.FormatLabelSelector(c.Selector))
+		}
+		msg = missingTargetMsg.String()
+	}
+
 	// If a check is missing it's kind, just mark it as completed
 	if c.TargetRef.Kind == "" {
 		ok = true
```

Comment:
Does this cause a gap?
If no items and targetRef.kind is empty, we set ok and discard `msg` ( in the if statement below, couldn't highlight the whole block).

Reply:
Would it be more clear if I moved the message construction after the potential return below?